clients/fides-js/src/docs/fides.ts

```diff
@@ -407,6 +421,76 @@ export interface Fides {
    */
   identity: Record<string, string>;
 
+  /**
+   * Updates user consent preferences with either a `consent` object or `fidesString`.
+   * If both are provided, `fidesString` takes priority.
+   *
+   * @example
+   * Update consent using notice keys and boolean values:
+   * ```ts
+   * Fides.updateConsent({
+   *   consent: {
+   *     data_sales_and_sharing: false,
+   *     analytics: true
+   *   }
+   * });
+   * ```
+   *
+   * @example
+   * Update consent using string values instead of booleans:
+   * ```ts
+   * Fides.updateConsent({
+   *   consent: {
+   *     data_sales_and_sharing: "opt_out",
+   *     analytics: "opt_in",
+   *     terms_of_service: "acknowledge"
+   *   }
+   * });
+   * ```
+   *
+   * @example
+   * Update consent using a fidesString:
+   * ```ts
+   * Fides.updateConsent({
+   *   fidesString: ",,,eyJkYXRhX3NhbGVzX2FuZF9zaGFyaW5nIjowLCJhbmFseXRpY3MiOjF9"
+   * });
+   * ```
+   *
+   * @example
+   * Control validation behavior:
+   * ```ts
+   * // With validation="warn" - logs warnings but doesn't throw errors
+   * Fides.updateConsent({
+   *   consent: { notice_key: invalidValue },
+   *   validation: "warn"
+   * });
+   *
+   * // With validation="ignore" - silently accepts invalid values
+   * Fides.updateConsent({
+   *   consent: { notice_key: invalidValue },
+   *   validation: "ignore"
+   * });
+   * ```
+   *
+   * @param options - Options for updating consent
+   * @param options.consent - Object mapping notice keys to consent values:
+   *   - Boolean values: `true` (opt-in/consent granted) or `false` (opt-out/consent declined)
+   *   - String values:
+   *     - `"opt_in"` - user has explicitly opted in to this notice
+   *     - `"opt_out"` - user has explicitly opted out of this notice
+   *     - `"acknowledge"` - ONLY valid for notices with "notice_only" consent mechanism
```

Comment:
do we also need to update this with `"not_applicable"` option too?

Reply:
Good callout. No, we don't want to support non_applicable but we do want to create an error specifically for that purpose that I forgot to add (see Neville's comment in JIRA https://ethyca.atlassian.net/browse/ENG-439?focusedCommentId=53324)